### PR TITLE
display CommandParsingException gracefully

### PR DESCRIPTION
**Pull request by jonsequitur**
This set of changes handles `CommandParsingException` gracefully (so as not to show the user a stack trace) and generalizes graceful exception display somewhat away from being type-specific.

It specifically addresses #5470.

### CODE REVIEW

**Comment by jonsequitur** [comment]
@livarcocc @piotrpMSFT @eerhardt @MichaelSimons

File: test/dotnet-new.Tests/NewCommandTests.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+using System;
+using System.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.New.Tests
+{
+    public class NewCommandTests
+    {
+		private readonly ITestOutputHelper output;
```
Comment (eerhardt) [inline test/dotnet-new.Tests/NewCommandTests.cs]: (double nit) spacing and private fields should start with `_`.  Then you can get rid of `this.` in the constructor too.  :)

File: test/dotnet-new.Tests/NewCommandTests.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+		public NewCommandTests(ITestOutputHelper output)
+		{
+			this.output = output;
+		}
+
+        [Fact]
+        public void WhenSwitchIsSkippedThenItPrintsError()
+        {
+            var cmd = new DotnetCommand().Execute("new Web1.1");
```
Comment (eerhardt) [inline test/dotnet-new.Tests/NewCommandTests.cs]: Since we know that "dotnet new3" is coming in and replacing "dotnet new", it feels like we should use a different command and switch here.

File: test/dotnet-new.Tests/NewCommandTests.cs (hunk 1 above)
Comment (jonsequitur) [inline test/dotnet-new.Tests/NewCommandTests.cs]: Old team's convention. Working on changing that habit. 👍

File: test/dotnet-new.Tests/NewCommandTests.cs (hunk 2 above)
Comment (jonsequitur) [inline test/dotnet-new.Tests/NewCommandTests.cs]: The bug being fixed is specific to the "new" command regardless of the underlying implementation, which is why I wrote this test. Are you concerned we'll lose coverage when new3 is swapped in? Some other concern?

We've been discussing a more consolidated approach to command line parsing and validation. That is out of scope for this change, though.

Comment (eerhardt) [inline test/dotnet-new.Tests/NewCommandTests.cs]: My concern is that it will just add more work for #5430 to get merged in.  Where if we picked a different command and switch, we wouldn't have to rewrite this test when #5430 is merged.

**Comment by jonsequitur** [comment]
@dotnet-bot test Windows_NT x86 Debug Build

**Comment by jonsequitur** [comment]
@dotnet-bot test Ubuntu16.04 x64 Debug Build

File: test/dotnet-new.Tests/NewCommandTests.cs (hunk 2 above)
Comment (jonsequitur) [inline test/dotnet-new.Tests/NewCommandTests.cs]: This test is not specific to New or New3. Is it an invalid test of New3?

Comment (eerhardt) [inline test/dotnet-new.Tests/NewCommandTests.cs]: Yes, New3 doesn't take the '-t'. So 'dotnet new3 Web1.1' is actually a valid command line.

**Comment by jonsequitur** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build

File: src/Microsoft.DotNet.Cli.Utils/ExceptionExtensions.cs [hunk 3]
```diff
@@ -3,13 +3,18 @@
...
     {
         public static void ReportAsWarning(this Exception e)
         {
             Reporter.Verbose.WriteLine($"Warning: Ignoring exception: {e.ToString().Yellow()}");
         }
+
+        public static bool ShouldBeDisplayedAsError(this Exception e) =>
+            e.Data.Contains(CLI_User_Displayed_Exception);
+
+        internal const string CLI_User_Displayed_Exception = "CLI_User_Displayed_Exception";
```
Comment (krwq) [inline src/Microsoft.DotNet.Cli.Utils/ExceptionExtensions.cs]: this feels like it should be an enum value

**Comment by jonsequitur** [comment]
@dotnet-bot test Windows_NT x86 Debug Build
